apps/scan/backend/src/electrical_testing/loops/card_read_and_usb_drive_write_loop.ts

```diff
@@ -0,0 +1,75 @@
+import * as fs from 'node:fs/promises';
+import * as path from 'node:path';
+import { assert, err, ok, Result, sleep } from '@votingworks/basics';
+import { LogEventId } from '@votingworks/logging';
+import { constructAuthMachineState } from '../../util/auth';
+import { ServerContext } from '../context';
+import { resultToString } from '../utils';
+
+export const CARD_READ_AND_USB_DRIVE_WRITE_INTERVAL_SECONDS = 5;
+export const USB_DRIVE_FILE_NAME = 'electrical-testing.txt';
+
+export async function runCardReadAndUsbDriveWriteLoop({
+  auth,
+  cardTask,
+  logger,
+  usbDrive,
+  usbDriveTask,
+  workspace,
+}: ServerContext): Promise<void> {
+  void cardTask.stopped().then((reason) => {
+    void logger.log(LogEventId.BackgroundTaskCancelled, 'system', {
+      message: `Card read loop stopping. Reason: ${reason}`,
+    });
+  });
+
+  void usbDriveTask.stopped().then((reason) => {
+    void logger.log(LogEventId.BackgroundTaskCancelled, 'system', {
+      message: `USB drive write loop stopping. Reason: ${reason}`,
+    });
+  });
+
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    // Exit the loop if both tasks are stopped.
+    if (cardTask.isStopped() && usbDriveTask.isStopped()) {
+      break;
+    }
+
+    // Wait for at least one task to be running.
+    await Promise.race([cardTask.running(), usbDriveTask.running()]);
+
+    if (cardTask.isRunning()) {
```

Comment:
Took me a bit of time to grok the difference between `.running()` and `.isRunning()`. Looking at the implementation of `TaskController` cleared my confusion. Maybe some name changes might make the external interface easier to understand without looking at the implementation, e.g.,

```ts
cardTask.isRunning()

await cardTask.waitUntilIsRunning()
```